lib/puppet.rb

```diff
@@ -42,11 +42,8 @@ module Puppet
   require 'puppet/environments'
 
   class << self
-    Puppet::GettextConfig.create_text_domain('production')
-    locale_dir = Puppet::GettextConfig.puppet_locale_path
-    if Puppet::GettextConfig.load_translations('puppet', locale_dir, Puppet::GettextConfig.translation_mode(locale_dir))
-      Puppet::GettextConfig.set_locale(Locale.current.language)
-    end
+    Puppet::GettextConfig.set_locale(Locale.current.language)
```

Comment:
We can't disable this initial step if i18n is turned off, because the settings telling us so haven't been loaded yet.

Reply:
Understood.

Reply:
Can we load this in [`Puppet::Settings.initialize_global_settings`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/settings.rb#L258-L270) instead, after parsing the CLI and puppet.conf? Ideally, I'd like to not load any `Gettext` stuff if `disable_i18n` is true.

Reply:
Unfortunately, there are many strings in both those places that have been marked for translation. We could theoretically unmark those strings (if we think we can reliably find them all), though communicating that distinction to anyone who wants to write and translate their own settings and CLI extensions might get tricky. Alternatively I suppose we could try to stub the translation functions before gettext gets loaded, though since all that happens at a couple different super [global|https://github.com/puppetlabs/puppet/blob/master/lib/puppet/gettext/config.rb#L17-L19] [levels|https://github.com/puppetlabs/puppet/blob/master/lib/puppet/gettext/config.rb#L25] (the first adds the methods to `Object`, because they were not written in a way that makes it possible to add them unqualified otherwise. The second adds them specifically to `main`, which is safer, because we have control over the module nesting), I'm a little worried about weird interactions when you then try to load the real translations functions.

Reply:
Nevermind, I think I found a way around it, that will also fix the unit tests. Update incoming.